### PR TITLE
Allow guests to send state events

### PR DESCRIPTION
**Pull request by robertlong**
This PR implements part of https://github.com/matrix-org/matrix-doc/pull/3419 needed to make group calls work. I have no idea if this is the proper way to do this and MSC3419 also has requirements to allow for more guest event types if they have the permission to do so. So we probably want to have a bigger discussion around this.

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

**Comment by clokep** [comment]
@robertlong Could you add a `changelog.d/11378.feature` file that explains what this does (and so that CI will run)? Maybe something like:

```markdown
Allow guests to send state events per [MSC3419](https://github.com/matrix-org/matrix-doc/pull/3419).
```

Or something like that?

**Comment by robertlong** [comment]
Yup, added it. Thanks @clokep! Also, I'm still unsure if this is good enough to push. I don't fully understand the implications of this change.

**Comment by erikjohnston** [comment]
This looks good. I don't think we can merge it as is until the MSC lands. Though we could land this if we put it behind an experimental feature flag, if we wanted to test this on a real deployment somewhere

**Review by erikjohnston — COMMENTED** [review]
LGTM, but blocked on MSC landing

**Comment by clokep** [comment]
[MSC3419](https://github.com/matrix-org/matrix-doc/pull/3419) has been merged.

**Review by reivilibre — APPROVED** [review]

**Pull request merged**
